app/js/translations.js

```diff
@@ -25,45 +15,49 @@ module.exports = {
         'locales/missing-__lng__.json'
       ),
       sendMissingTo: 'fallback',
-      fallbackLng: (options != null ? options.defaultLng : undefined) || 'en',
+      fallbackLng: options.defaultLng || 'en',
       detectLngFromHeaders: true,
       useCookie: false,
       preload: availableLngs,
       supportedLngs: availableLngs
     })
-    const setLangBasedOnDomainMiddlewear = function(req, res, next) {
+
+    function setLangBasedOnDomainMiddleware(req, res, next) {
+      // Determine language from subdomain
       const { host } = req.headers
       if (host == null) {
         return next()
       }
-      const parts = host.split(/[.-]/)
-      const subdomain = parts[0]
-      const lang = __guard__(
-        __guard__(
-          options != null ? options.subdomainLang : undefined,
-          x1 => x1[subdomain]
-        ),
-        x => x.lngCode
-      )
-      if (req.originalUrl.indexOf('setLng') === -1 && lang != null) {
+      const [subdomain] = host.split(/[.-]/)
+      const lang = subdomainLang[subdomain]
+        ? subdomainLang[subdomain].lngCode
+        : null
+
+      // Unless setLng query param is set, use subdomain lang
+      if (!req.originalUrl.includes('setLng') && lang != null) {
         req.i18n.setLng(lang)
       }
+
+      // If the set language (req.lng) is different from the language
+      // automatically determined by the i18next middleware (req.language), then
+      // set flag which will show a banner offering to switch to the appropriate
+      // library
```

Comment:
🤦  just realised that this is typo.

```suggestion
      // language
```